pkg/plugin/admissioncontrollers/webhook/admission_controller.go

```diff
@@ -138,25 +138,7 @@ func updateLabels(pod *v1.Pod, patch []patchOperation) []patchOperation {
 
 	if _, ok := existingLabels[common.SparkLabelAppID]; !ok {
 		if _, ok := existingLabels[common.LabelApplicationID]; !ok {
-			// if app id not exist, generate one
-			// the generated ID is using [PodNamespace]_[PodName]_[Timestamp] naming convention.
-			// some admission controllers have strict checks of the length/format of each labels,
-			// this convention keeps the name tidy and short.
-			podNamespace := "default"
-			if pod.Namespace != "" {
-				podNamespace = pod.Namespace
-			}
-
-			// pod's name can be generated, if name is not explicitly specified
-			// look for generateName instead
-			podName := "unknown"
-			if pod.Name != "" {
-				podName = pod.Name
-			} else if pod.GenerateName != "" {
-				podName = pod.GenerateName
-			}
-
-			generatedID := fmt.Sprintf("%s_%s_%d", podNamespace, podName, time.Now().Unix())
+			generatedID := fmt.Sprintf("__app_%d", time.Now().UnixNano())
```

Comment:
I do not understand the change. We are going from using the pod name or generate name combined with a timestamp to just `app` and a timestamp. It would make tracking pods on the YuniKorn side really difficult.

Can we not use a length limited label setup still leveraging the name or generate name?
* remove the podNamespace (not making it more unique)
* limit the size of podName to 50 char in the printed text
* use a `_` as separator
* use the millis since 1970 (13 characters) if seconds is not good enough
```
	timestamp := strconv.FormatInt(time.Now().UnixNano(), 10)
	generatedID := fmt.Sprintf("%.50s_%.13s", podName, timestamp)
```
This will give a guaranteed maximum of 64 chars. We can even tweak that down if you think that is needed.

Reply:
One of the main reasons to have this change, not just to keep it < 64 char. Is the fact during the webhook phase, pod namespace, pod name could both be empty (in many cases). My suggestion is to use a general generated ID just with a timestamp, but get more info from attributes https://issues.apache.org/jira/browse/YUNIKORN-55.

Reply:
Why throw out all the info we have and then try to substitute it with some other info?
If we have the detail we should use it. The change will limit the label to a maximum of 64 char.

If the build up `podName` turns out to be empty we should substitute at that point instead of the  generic rule.

Reply:
To me @wilfred-s suggestion make sense, assuming podName is not going to change. 
It is much more informative from debugging perspective for instance to have IDs with more details in it.

Reply:
Thanks for the review. I think that makes sense, we can do that.
In my latest commit, I use a very similar approach like @wilfred-s suggested, the only difference is to maintain a full timestamp to ensure the ID would be unique (pod name might be the same). UT added.  Please help to review, thanks!